openmdao/core/tests/test_expl_comp.py

```diff
@@ -671,6 +687,130 @@ def compute(self, inputs, outputs):
                                     hierarchical=True,
                                     print_arrays=True)
 
+    def test_simple_var_tags(self):
+        prob = om.Problem(RectangleCompWithTags())
+        prob.setup(check=False)
+        prob.run_model()
+
+        # Inputs no tags
+        inputs = prob.model.list_inputs(out_stream=None)
+        self.assertEqual(sorted(inputs), [
+            ('length', {'value': [1.]}),
+            ('width', {'value': [1.]}),
+        ])
+
+        # Inputs with tags
+        inputs = prob.model.list_inputs(out_stream=None, tags="tag1")
+        self.assertEqual(sorted(inputs), [
+            ('length', {'value': [1.]}),
+        ])
+
+        # Inputs with multiple tags
+        inputs = prob.model.list_inputs(out_stream=None, tags=["tag1", "tag3"])
+        self.assertEqual(sorted(inputs), [
+            ('length', {'value': [1.]}),
+        ])
+        inputs = prob.model.list_inputs(out_stream=None, tags=["tag1", "tag2"])
+        self.assertEqual(sorted(inputs), [
+            ('length', {'value': [1.]}),
+            ('width', {'value': [1.]}),
+        ])
+
+        # Inputs with tag that does not match
+        inputs = prob.model.list_inputs(out_stream=None, tags="tag3")
+        self.assertEqual(sorted(inputs), [])
+
+        # Outputs no tags
+        outputs = prob.model.list_outputs(out_stream=None)
+        self.assertEqual(sorted(outputs), [
+            ('area', {'value': [1.]}),
+        ])
+
+        # Outputs with tags
+        outputs = prob.model.list_outputs(out_stream=None, tags="tag1")
+        self.assertEqual(sorted(outputs), [
+            ('area', {'value': [1.]}),
+        ])
+
+        # Outputs with multiple tags
+        outputs = prob.model.list_outputs(out_stream=None, tags=["tag1", "tag3"])
+        self.assertEqual(sorted(outputs), [
+            ('area', {'value': [1.]}),
+        ])
+
+        # Outputs with tag that does not match
+        outputs = prob.model.list_outputs(out_stream=None, tags="tag3")
+        self.assertEqual(sorted(outputs), [])
+
+    def test_feature_simple_var_tags(self):
+        from openmdao.api import Problem, ExplicitComponent
+
+        class RectangleCompWithTags(ExplicitComponent):
```

Comment:
it would be nice if we had a more realistic example for the docs demonstrating how/why this feature is useful, rather than just these arbitrary tags...